modin/pandas/test/test_groupby.py

```diff
@@ -1346,15 +1349,30 @@ def test_multi_column_groupby():
         modin_df.groupby(by, axis=1).count()
 
 
-def sort_index_if_experimental_groupby(*dfs):
+def sort_if_experimental_groupby(*dfs):
     """
     This method should be applied before comparing results of ``groupby.transform`` as
     the experimental implementation changes the order of rows for that:
     https://github.com/modin-project/modin/issues/5924
     """
+    result = dfs
     if RangePartitioningGroupby.get():
-        return tuple(df.sort_index() for df in dfs)
```

Comment:
previously, the function was only sorting by index, however in case of the low amount of unique values in the index this approach may not work well, so sorting on column values as it appears to be more stable for dataframes